streams/src/test/java/org/apache/kafka/streams/integration/AbstractJoinIntegrationTest.java

```diff
@@ -166,35 +138,12 @@ void prepareEnvironment() throws InterruptedException {
         }
 
         STREAMS_CONFIG.put(StreamsConfig.STATE_DIR_CONFIG, testFolder.getRoot().getPath());
-
-        producer = new KafkaProducer<>(PRODUCER_CONFIG);
     }
-
     @After
     public void cleanup() throws InterruptedException {
-        producer.close(Duration.ofMillis(0));
         CLUSTER.deleteAllTopicsAndWait(120000);
     }
 
-    private void checkResult(final String outputTopic, final List<KeyValueTimestamp<Long, String>> expectedResult) throws InterruptedException {
-        IntegrationTestUtils.verifyKeyValueTimestamps(RESULT_CONSUMER_CONFIG, outputTopic, expectedResult);
-    }
-
-    private void checkResult(final String outputTopic, final KeyValueTimestamp<Long, String> expectedFinalResult, final int expectedTotalNumRecords) throws InterruptedException {
-        final List<KeyValueTimestamp<Long, String>> result =
-            IntegrationTestUtils.waitUntilMinKeyValueWithTimestampRecordsReceived(RESULT_CONSUMER_CONFIG, outputTopic, expectedTotalNumRecords, 30 * 1000L);
-        assertThat(result.get(result.size() - 1), is(expectedFinalResult));
-    }
-
-    /*
-     * Runs the actual test. Checks the result after each input record to ensure fixed processing order.
-     * If an input tuple does not trigger any result, "expectedResult" should contain a "null" entry
-     */
-    void runTest(final List<List<KeyValueTimestamp<Long, String>>> expectedResult) throws Exception {
-        runTest(expectedResult, null);
-    }
-
-
     void runTestWithDriver(final List<List<TestRecord<Long, String>>> expectedResult) {
```

Comment:
prop: Would it make sense to also have an overload with just a flat list, i.e.,
```
void runTestWithDriver(final List<TestRecord<Long, String>> expectedResult)
```
Maybe it would simplify the code of some of the tests. Hopefully, you can share some of the code in the overloads.